TPP.Core/Commands/Definitions/ModerationCommands.cs

```diff
@@ -162,4 +167,59 @@ private async Task<CommandResult> CheckTimeout(CommandContext context)
                 : $"{targetUser.Name} is not timed out. {infoText}"
         };
     }
+
+    private async Task<CommandResult> SetAppealCooldown(CommandContext context)
+    {
+        (User targetUser, TimeSpan timeSpan) =
+             await context.ParseArgs<User, TimeSpan>();
+        Duration duration = Duration.FromTimeSpan(timeSpan);
+        return new CommandResult
+        {
+            Response = await _moderationService.SetAppealCooldown(context.Message.User, targetUser, duration) switch
+            {
+                SetAppealCooldownResult.Ok => $"{targetUser.Name} is now eligible to appeal at {_clock.GetCurrentInstant() + duration}.",
+                SetAppealCooldownResult.UserNotBanned => $"{targetUser.Name} is not banned.",
+                SetAppealCooldownResult.AlreadyPermanent => throw new InvalidOperationException("Unexpected AlreadyPermanent repsonse"),
+            }
+        };
+    }
+
+    private async Task<CommandResult> SetUnappealable(CommandContext context)
+    {
+        User targetUser = await context.ParseArgs<User>();
+        return new CommandResult
+        {
+            Response = await _moderationService.SetUnappealable(context.Message.User, targetUser) switch
+            {
+                SetAppealCooldownResult.Ok => $"{targetUser.Name} is now ineligible to appeal their ban.",
+                SetAppealCooldownResult.UserNotBanned => $"{targetUser.Name} is not banned.",
+                SetAppealCooldownResult.AlreadyPermanent => $"{targetUser.Name} is already ineligible to appeal.",
+            }
+        };
+    }
+
+    private async Task<CommandResult> CheckAppealCooldown(CommandContext context)
+    {
+        User targetUser = await context.ParseArgs<User>();
+        if (!targetUser.Banned)
+            return new CommandResult { Response = $"{targetUser.Name} is not banned." };
+
+        AppealCooldownLog? recentLog = await _appealCooldownRepo.FindMostRecent(targetUser.Id);
+        string? issuerName = recentLog?.IssuerUserId == null
+            ? "<automated>"
+            : (await _userRepo.FindById(recentLog.IssuerUserId))?.Name;
+        string infoText = recentLog == null
+            ? "No logs available."
+            : $"Last action was {recentLog.Type} by {issuerName} " +
+              $"at {recentLog.Timestamp}";
+        if (recentLog?.Duration != null) infoText += $" for {recentLog.Duration.Value.ToTimeSpan().ToHumanReadable()}";
+
+        return new CommandResult { Response =
+            targetUser.AppealDate is null
+            ? $"{targetUser.Name} is not eligible to appeal. {infoText}"
+            : targetUser.AppealDate < _clock.GetCurrentInstant()
+            ? $"{targetUser.Name} is eligible to appeal now. {infoText}"
+            : $"{targetUser.Name} will be eligible to appeal on {targetUser.AppealDate}. {infoText}"
+        };
```

Comment:
not sure if `dotnet format` would like it, but making the ternary nesting a bit more visible with indentation might help a bit:
```suggestion
        return new CommandResult { Response = 
            targetUser.AppealDate is null
                ? $"{targetUser.Name} is not eligible to appeal. {infoText}"
                : targetUser.AppealDate < _clock.GetCurrentInstant()
                    ? $"{targetUser.Name} is eligible to appeal now. {infoText}"
                    : $"{targetUser.Name} will be eligible to appeal on {targetUser.AppealDate}. {infoText}"
        };
```
Or, given the infoText gets appended anyway, maybe:
```suggestion
        return new CommandResult { Response = (
            targetUser.AppealDate is null
                ? $"{targetUser.Name} is not eligible to appeal."
                : targetUser.AppealDate < _clock.GetCurrentInstant()
                    ? $"{targetUser.Name} is eligible to appeal now."
                    : $"{targetUser.Name} will be eligible to appeal on {targetUser.AppealDate}."
            ) + " " + infoText
        };
```